service/history/transferQueueProcessor.go

```diff
@@ -361,15 +361,34 @@ func (t *transferQueueProcessorImpl) processDecisionTask(task *persistence.Trans
 		WorkflowId: common.StringPtr(task.WorkflowID),
 		RunId:      common.StringPtr(task.RunID),
 	}
-
 	taskList := &workflow.TaskList{
 		Name: &task.TaskList,
 	}
+
+	// get workflow timeout
```

Comment:
Could this value to be passed through transfer task instead of requiring loading of a mutable state?

Reply:
Loading mutable state here is not an additional call to DB, so I kept this solution, and optimized the call to avoid duplicate call.